### PR TITLE
[android] added all preferences to the newly created settings activity

### PR DESCRIPTION
**Pull request by marcelinkaaa**
@aldenml Just added all the preferences in the discussed order.

Plan of action:
1) adjust some few remaining custom layouts we use for preferences, add new styles
2) work on making preferences functional ( with @grzesiekrzaca ):
    a) BitTorrent connection, Wifi networks only, Select Storage
    b) Toggle state and search settings
    c) Torrent preferences
    d) Other
3) work on the single and double pane view of settings based on screen size

### CODE REVIEW

File: android/res/xml/settings_search.xml [hunk 1]
```diff
@@ -18,84 +18,55 @@
...
-
-    <!--<com.frostwire.android.gui.activities.ToggleAllSearchEnginesPreference
-        android:key="frostwire.prefs.search.preference_category.select_all"
-        android:title="@string/select_deselect_all" />-->
+    <!--<com.frostwire.android.gui.activities.ToggleAllSearchEnginesPreference-->
+        <!--android:key="frostwire.prefs.search.preference_category.select_all"-->
+        <!--android:title="@string/select_deselect_all" />-->
     <CheckBoxPreference
         android:key="frostwire.prefs.search.use_archiveorg"
-        android:layout="@layout/view_preference_simple_action"
```
Comment (aldenml) [inline android/res/xml/settings_search.xml]: 👍

Comment (marcelinkaaa) [inline android/res/xml/settings_search.xml]: yes - material now does the same thing :)

File: android/res/xml/settings_application.xml [hunk 2]
```diff
@@ -18,35 +18,56 @@
  */
 -->
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
+    <PreferenceCategory android:key="frostwire.prefs.general"
```
Comment (aldenml) [inline android/res/xml/settings_application.xml]: just a note: this `key` might be not necessary after all

**Comment by gubatron** [comment]
1) ~~adjust some few remaining custom layouts we use for preferences, add new styles~~

File: android/res/xml/settings_application.xml (hunk 2 above)
Comment (gubatron) [inline android/res/xml/settings_application.xml]: I think it's useful to get a hold of the category in java-land, at least before material design

```java
    private void setupTorrentMaxDownloadSpeed(final BTEngine e) {
        NumberPickerPreference pickerPref = (NumberPickerPreference) findPreference(Constants.PREF_KEY_TORRENT_MAX_DOWNLOAD_SPEED);
...
```

**Pull request merged**